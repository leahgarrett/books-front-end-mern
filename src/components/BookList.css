
.book-list {
    padding: 20px 0;
}
.book-list  div {
    padding: 7px;
    margin: 5px 0;
    border: 1px solid #ccc;
    color: white;

}
.comedy {
    background-color: darkmagenta;
    
  }

  .drama {
      background-color: maroon;
  }

  .action {
    background-color: darkblue;
  }

  .thriller {
    background-color: darkgreen;
  }