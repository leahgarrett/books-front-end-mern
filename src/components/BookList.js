import React, { Component } from 'react';
import './BookList.css';

class BookList extends Component {
  constructor(props) {
      super(props);
      this.state = { 
          books: []
      };
    }

    handleClickBook = (book) => {
      this.props.handleEditClick(book)
    }


  render(){
      return (
        <div className="book-list">
        {this.props.books.map((item, index) => (
            <div onClick={() => this.handleClickBook(item)} key={index} className={item.genre}>
            {item.genre}: {item.title} by {item.author} (rrp <span className="price">${item.price.toFixed(2)}</span>) (id: {item.id})
            </div>
        ))}
      </div>
      );
    }
}


  export default BookList;

