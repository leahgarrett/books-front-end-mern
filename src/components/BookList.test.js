import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";
import BookList from "./BookList";

describe("<BookList />", () => {
    const books3 = [{
    author: "Ms. Remington Fahey",
    genre: "comedy",
    id: 1,
    price: 53.7,
    title: "Legacy Intelligent",
    _id: "5d19cd74a0c3e2c7f68ff800"
    },
    {
    author: "Isaac Steuber",
    genre: "thriller",
    id: 2,
    price: 20.99,
    title: "Incredible Incredible Plastic Computer",
    _id: "5d19cd74a0c3e2c7f68ff801"
    },
    {
    author: "Kennedy Howell",
    genre: "thriller",
    id: 3,
    price: 68.61,
    title: "compressing Soft mint green",
    _id: "5d19cd74a0c3e2c7f68ff802"
    }
    ];

    it("renders component when there are no books", () => {
    const books = [];
    const wrapper = render(
        <BookList books={books} handleEditClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
    });

    it("renders component when there are three books", () => {
    
    const wrapper = shallow(
        <BookList books={books3} handleEditClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
    });

    it("correctly formats the price to two decimal places", () => {
      const wrapper = mount(
        <BookList books={[books3[0]]} handleEditClick={() => {}} />
      );
      const text = wrapper.find(".price").text();
  
      expect(text).toEqual('$53.70');
    });

    it("contains the genre", () => {
      const wrapper = mount(
        <BookList books={[books3[0]]} handleEditClick={() => {}} />
      );
      const text = wrapper.find(".comedy").text();
      expect(text.includes('comedy')).toEqual(true);
    });

    it("contains the book title", () => {
      const wrapper = mount(
        <BookList books={[books3[0]]} handleEditClick={() => {}} />
      );
      const text = wrapper.find(".comedy").text();
      expect(text.includes('Legacy Intelligent')).toEqual(true);
    });


    it("calls the edit method on click", () => {
      const spy = sinon.spy();
      const wrapper = mount(
        <BookList books={[books3[0]]} handleEditClick={() => {spy()}} />
      );
      
      wrapper.find(".comedy").simulate("click");
      expect(spy.calledOnce).toBe(true);
    });
});