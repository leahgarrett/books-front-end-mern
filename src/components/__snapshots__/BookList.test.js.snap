// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookList /> renders component when there are no books 1`] = `
<div
  class="book-list"
/>
`;

exports[`<BookList /> renders component when there are three books 1`] = `
<div
  className="book-list"
>
  <div
    className="comedy"
    key="0"
    onClick={[Function]}
  >
    comedy
    : 
    Legacy Intelligent
     by 
    Ms. Remington Fahey
     (rrp 
    <span
      className="price"
    >
      $
      53.70
    </span>
    ) (id: 
    1
    )
  </div>
  <div
    className="thriller"
    key="1"
    onClick={[Function]}
  >
    thriller
    : 
    Incredible Incredible Plastic Computer
     by 
    Isaac Steuber
     (rrp 
    <span
      className="price"
    >
      $
      20.99
    </span>
    ) (id: 
    2
    )
  </div>
  <div
    className="thriller"
    key="2"
    onClick={[Function]}
  >
    thriller
    : 
    compressing Soft mint green
     by 
    Kennedy Howell
     (rrp 
    <span
      className="price"
    >
      $
      68.61
    </span>
    ) (id: 
    3
    )
  </div>
</div>
`;
